base/reflection.jl

```diff
@@ -1035,7 +1035,7 @@ Returns a tuple `(filename,line)` giving the location of a `Method` definition.
 function functionloc(m::Method)
     ln = m.line
     if ln <= 0
-        error("could not determine location of method definition")
+        throw(ArgumentError("could not determine location of method definition"))
```

Comment:
:+1: To most of these changes. I'm not sure this one is an argument error though; it doesn't really seem like the caller's fault.

Reply:
You are right. Not sure when we have this kind of case. Any suggestion for a more specific exception type rather than throwing a general ExceptionError?

Reply:
Sometimes an `ErrorException` is the best fit, might be so here.